src/ILCompiler.RyuJit/src/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -484,5 +485,109 @@ private InfoAccessType constructStringLiteral(CORINFO_MODULE_STRUCT_* module, md
+                    if (_localSlotToInfoMap.TryGetValue(slotNumber, out ilVar))
+                    {
+                        updateDebugVarInfo(debugVars, ilVar.Name, false, nativeVarInfo);
+                    }
+                }
+            }
+
+            i = 0;
+            _debugVarInfos = new DebugVarInfo[debugVars.Count];
+            foreach (var debugVar in debugVars)
+            {
+                _debugVarInfos[i] = debugVar.Value;
+                i++;
+            }
+        }
+
+        // Create a DebugLocInfo which is a table from native offset to source line.
+        // using native to il offset (pMap) and il to source line (_sequencePoints).
+        private void setBoundaries(CORINFO_METHOD_STRUCT_* ftn, uint cMap, OffsetMapping* pMap)
+        {
+            Debug.Assert(_debugLocInfos == null);
+            // No interest if sequencePoints is not populated before.
+            if (_sequencePoints == null)
+            {
+                return;
+            }
+
+            int largestILOffset = 0; // All epiloges point to the largest IL offset.
+            for (int i = 0; i < cMap; i++)
+            {
+                OffsetMapping nativeToILInfo = pMap[i];
+                int currectILOffset = (int)nativeToILInfo.ilOffset;
+                if (currectILOffset > largestILOffset) // Special offsets are negative.
+                {
+                    largestILOffset = currectILOffset;
+                }
+            }
+
+            int previousNativeOffset = -1;
+            List<DebugLocInfo> debugLocInfos = new List<DebugLocInfo>();
+            for (int i = 0; i < cMap; i++)
+            {
+                OffsetMapping nativeToILInfo = pMap[i];
+                int ilOffset = (int)nativeToILInfo.ilOffset;
+                int nativeOffset = (int)pMap[i].nativeOffset;
+                if (nativeOffset == previousNativeOffset)
+                {
+                    // Save the first one, skip others.
+                    continue;
+                }
+                switch (ilOffset)
+                {
+                    case (int)MappingTypes.PROLOG:
+                        ilOffset = 0;
+                        break;
+                    case (int)MappingTypes.EPILOG:
+                        ilOffset = largestILOffset;
+                        break;
+                    case (int)MappingTypes.NO_MAPPING:
+                        continue;
+                }
+
+                NativeLocInfo nativeLocInfo = new NativeLocInfo { nativeOffset = nativeToILInfo.nativeOffset, ilOffset = nativeToILInfo.ilOffset, source = (uint)nativeToILInfo.source };
+
+                SequencePoint s;
+                if (_sequencePoints.TryGetValue((int)ilOffset, out s))
+                {
+                    Debug.Assert(s.Document != null);
+                    DebugLocInfo loc = new DebugLocInfo(nativeOffset, s.Document, nativeLocInfo, s.LineNumber);
```

Comment:
I think column number is getting lost somewhere along the way

Reply:
Not sure why - that's how CoreRT works today

Reply:
I'd think that's just a bug. It would be good not to lose columns for things like multi-statement lambdas or properties.